### ISSUE TITLE
TypeError 'numpy._DTypeMeta' object is not subscriptable

### ISSUE THREAD

**Issue by lukegwoolley** [ERGO-Code/HiGHS#2025]
Happens on both latest numpy version and numpy 1.21.5
![highspy numpy error](https://github.com/user-attachments/assets/7ce33500-2b02-4101-9bb9-3e4d1f490362)

Error sits in this line:
File /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/highspy/highs.py:21
     19 # backwards typing support information for HighspyArray
     20 if sys.version_info >= (3, 9):
---> 21     ndarray_object_type = np.ndarray[Any, np.dtype[np.object_]]

**Comment by jajhall** [comment]
Any thoughts on how to fix this @mathgeekcoder ?

**Comment by mathgeekcoder** [comment]
Oh interesting.  Python typing support seems a little troublesome.  Since I can't reproduce, @lukegwoolley, can you please try to modify your highs.py, and see if this works for you? 

```python
# backwards typing support information for HighspyArray
if sys.version_info >= (3, 9):
    from typing import TypeAlias
    obj_type: TypeAlias = 'np.dtype[np.object_]'
    ndarray_object_type = np.ndarray[Any, obj_type]
else:
    ndarray_object_type = np.ndarray
```

If that works, I can make a new PR with some relevant fixes.

**Comment by lukegwoolley** [comment]
Hi @mathgeekcoder 

I had a play around with the code and I think the issue is the sys.version issue is wrong. 

The below code works for me in python 3.10 and 3.12 but I wasnt able to test for 3.11.

# backwards typing support information for HighspyArray
if sys.version_info >= (3, 11):
    ndarray_object_type = np.ndarray[Any, np.dtype[np.object_]]
else:
    ndarray_object_type = np.ndarray

**Comment by mathgeekcoder** [comment]
Thanks @lukegwoolley.  Did you happen to test the code I sent?  Can you confirm if it didn't work?

Updating that version check just disables the numpy typing for your python 3.10.   According to the docs, this typing should be supported in 3.10.

Also, do you know if there's anything special about your environment?  I'm wondering since this error wasn't picked up in our CI build - which also tests the python wheel on many different versions.

**Comment by lukegwoolley** [comment]
![image](https://github.com/user-attachments/assets/f41c5125-c789-4b2c-9c50-1c76132ea70d)

It errors out here - it might be due to the numpy version then as unfortunately databricks is where I am running it and it uses a very old numpy and pandas version. 

numpy is 1.21.5 

The previous version of highspy 1.7.3 works completely fine btw

**Comment by jajhall** [comment]
Closed by #2068

**Comment by mathgeekcoder** [comment]
Thanks for the confirmation and fix @quantresearch1!  Apologies for the delayed reply, I've been sick for the past couple weeks.
